gwcs/tags/selectortags.py

```diff
@@ -0,0 +1,121 @@
+        inputs_mapping = node.get('inputs_mapping', None)
+        if inputs_mapping is not None and not isinstance(inputs_mapping, models.Mapping):
+            raise TypeError("inputs_mapping must be an instance"
+                            "of astropy.modeling.models.Mapping.")
+        mapper = node['mapper']
+
+        if isinstance(mapper, NDArrayType):
+            if mapper.ndim != 2:
+                raise NotImplementedError(
+                    "GWCS currently only supports 2x2 masks ")
+            return LabelMapperArray(mapper, inputs_mapping)
+        else:
+            inputs = node.get('inputs', None)
+            labels = mapper.get('labels')
+            transforms = mapper.get('models')
+            dict_mapper = dict(zip(labels, transforms))
+            return LabelMapperDict(inputs, dict_mapper, inputs_mapping)
+
+    @classmethod
+    def to_tree_transform(cls, model, ctx):
+        node = OrderedDict()
+        if isinstance(model, LabelMapperArray):
+            node['mapper'] = model.mapper
+        if isinstance(model, (LabelMapperDict, LabelMapperRange)):
+            mapper = OrderedDict()
+            labels = list(model.mapper.keys())[:]
+
+            transforms = []
+            for k in labels:
+                transforms.append(model.mapper[k])
+            if not np.isscalar(labels[0]):
+                labels = [list(l) for l in labels]
+            mapper['labels'] = labels
+            mapper['models'] = transforms
+            node['mapper'] = mapper
+            node['inputs'] = list(model.inputs)
+        if model.inputs_mapping is not None:
+            node['inputs_mapping'] = model.inputs_mapping
+
+        return yamlutil.custom_tree_to_tagged_tree(node, ctx)
+
+    @classmethod
+    def assert_equal(cls, a, b):
+        # TODO: If models become comparable themselves, remove this.
+        assert (a.__class__ == b.__class__)
+        if isinstance(a.mapper, dict):
+            assert_array_equal(list(a.mapper.keys()), list(b.mapper.keys()))
+            assert (a.inputs == b.inputs)
+            assert (a.inputs_mapping.mapping == b.inputs_mapping.mapping)
+        else:
+            assert_array_equal(a.mapper, b.mapper)
+
+
+class RegionsSelectorType(TransformType):
+    name = "transform/regions_selector"
+    types = [RegionsSelector]
+
+    @classmethod
+    def from_tree_transform(cls, node, ctx):
+        inputs = node['inputs']
+        outputs = node['outputs']
+        mask = node['mask']
+        undefined_transform_value = node['undefined_transform_value']
+        sel = node['selector']
+
+        return RegionsSelector(inputs, outputs,
+                                        sel, mask, undefined_transform_value)
+
+    @classmethod
+    def to_tree_transform(cls, model, ctx):
+        selector = OrderedDict()
+        node = OrderedDict()
+        labels = list(model.selector.keys())
+        values = []
+        for l in labels:
+            values.append(model.selector[l])
```

Comment:
I want the `values` list to be in the same order as the `labels` list, i.e. they should match so that the dict can be restored. I just don't know if `six.itervalues` ensures that.